### PR TITLE
[CMAKE] Fix system boot crash when compiled as NT6+

### PR DESCRIPTION
**Pull request by RatinCN**
## Purpose

Fix system boot crashed with `STATUS_ILLEGAL_DLL_RELOCATION (0xC0000269)` when compiled with `-DDLL_EXPORT_VERSION=0x600`.

JIRA issue: None

## Proposed changes

Run `gen_baseaddress.py` to update the baseaddress. Test DLLs are not involved, lemme know if `-DENABLE_ROSTESTS=1` is needed.

Before:

After:

In Debug mode, another error logged:
```
ntoskrnl/mm/ARM3/sysldr.c:170) Loading: \SystemRoot\system32\drivers\blue.sys at F80FC000 with f pages
(ntoskrnl/mm/ARM3/sysldr.c:170) Loading: \SystemRoot\system32\drivers\msfs.sys at F80F1000 with b pages
(ntoskrnl/mm/ARM3/sysldr.c:170) Loading: \SystemRoot\system32\drivers\npfs.sys at F80CC000 with 25 pages
(ntoskrnl/mm/ARM3/sysldr.c:170) Loading: \SystemRoot\system32\drivers\ndisuio.sys at F80BC000 with 10 pages
(ntoskrnl/mm/ARM3/sysldr.c:170) Loading: \SystemRoot\system32\drivers\afd.sys at F8089000 with 33 pages
(ntoskrnl/io/iomgr/driver.c:83) Deleting driver object '\Driver\KdDriver'
(drivers/storage/port/scsiport/ioctl.c:542) unknown ioctl code: 0x4DC010
(drivers/storage/port/scsiport/ioctl.c:542) unknown ioctl code: 0x4D0010
WARNING:  RtlCreateTagHeap at sdk/lib/rtl/heap.c:4024 is UNIMPLEMENTED!
(ntoskrnl/io/iomgr/error.c:714) IoRaiseInformationalHardError: c0000102, '\pagefile.sys'

*** Assertion failed: ErrorStatus != STATUS_FILE_CORRUPT_ERROR
***   Source File: ../ntoskrnl/io/iomgr/error.c, line 716
```
which is not point the problem out, thanks DarkFire01 who give me hint to find the problem around base address, all are work fine now.

### CODE REVIEW

**Review by DarkFire01 — APPROVED** [review]

**Review by SergeGautherie — COMMENTED** [review]
What is the problem here? That 0x600 dlls need more space than 0x502 ones?
Shouldn't we use a separate set of files for each version?
If not, I think we should had a reminder comment, in the very files in git, that, from now on, they must be regenerated with 0x600 dlls only, in order to support both versions.
And I hope 0x600 builds do not miss any 0x502 filenames...

**Comment by RatinCN** [comment]
> What is the problem here? That 0x600 dlls need more space than 0x502 ones? Shouldn't we use a separate set of files for each version? If not, I think we should had a reminder comment, in the very files in git, that, from now on, they must be regenerated with 0x600 dlls only, in order to support both versions. And I hope 0x600 builds do not miss any 0x502 filenames...

Currently, yes, 0x600 needs more space.

File: sdk/cmake/baseaddress.cmake [hunk 1]
```diff
@@ -1,121 +1,121 @@
-# Generated from C:\Users\katahiromz\reactos-1\output-MinGW-i386
+# Generated from ../../output-MinGW-i386
```
[the remark this answers is not in the document]
Comment (RatinCN) [reply sdk/cmake/baseaddress.cmake:1]: I suggest to preserve the directory name `output-MinGW-i386` which tells the build configuration.

**Comment by HBelusca** [comment]
> What is the problem here? That 0x600 dlls need more space than 0x502 ones? Shouldn't we use a separate set of files for each version?

Perhaps indeed it would be of interest to use new "baseaddress[_msvc]_0600.cmake" files for such builds, until at least we are able to auto-generate these at build time.

**Review by HBelusca — APPROVED** [review]

**Review by cbialorucki — APPROVED** [review]

**Review by JoachimHenze — CHANGES_REQUESTED** [review]
Please restore "baseaddress_custom" which you deleted.

Here is the output of GCC dbg i386 0x502 buildbot after your changes:
image

**Comment by RatinCN** [comment]
> Please restore "baseaddress_custom" which you deleted. Please restore "baseaddress_notifyhook" which you deleted. and so on...
> 
> Here is the output of GCC dbg i386 0x502 buildbot after your changes: image
> 
> Proving that you can use the github actions to verify the correctness of your changes.

`-DENABLE_ROSTESTS=1` added now, test dlls are involved.

**Comment by JoachimHenze** [comment]
> `-DENABLE_ROSTESTS=1` added now, test dlls are involved.

Thanks, it improved. But still 5 (instead of formerly 8) are missing:
image

Please restore them manually. They were present before.
You can use the Windows calculator, place them at the bottom and just subtract their size from the latest base-address that the script generated.
Place a `# manually added` comment after them. So the next one will not run into the same problem again.

**Comment by RatinCN** [comment]
> > `-DENABLE_ROSTESTS=1` added now, test dlls are involved.
> 
> Thanks, it improved. But still 5 (instead of formerly 9) are missing: image
> 
> If also adding `--DENABLE_ROSAPPS=1` additionally wouldn't help to fix them yet, then please restore them manually. They were present before. You can use the Windows calculator, place them at the bottom and just subtract their size from the latest base-address that the script generated. Place a `# manually added` comment after them. So the next one will not run into the same problem again.

- `rshell` manually added at the bottom now.
- `vfd` and `notifyhook` not found, I checked git log in past 1 year, from [Commit 0fb580a8](https://github.com/reactos/reactos/commit/0fb580a8553e1028fb942ed8c061f9cd88d319e9) to now.
- `msi_custom` and `msi_selfreg` is `custom` and `selfreg`, seems they should be renamed in [Commit 6c93f3ca](https://github.com/reactos/reactos/commit/6c93f3ca6b95fa5e1806fc29295d3bc8ada979f9), but the name in this file didn't changed after that, both of this PR and [Commit 73d0d24d](https://github.com/reactos/reactos/commit/73d0d24d679101e6f7e69b02c7c50062b4f7d924) Katahiromz. I renamed them manually in this PR. Please @HBelusca have a check? The name generated by the script are still `custom` and `selfreg`, not `msi_custom` and `msi_selfreg`.

**Comment by RatinCN** [comment]
Build failure seems unrelated:
```
2024-05-13T09:02:49.7611568Z FAILED: dll/win32/ws2_32/CMakeFiles/ws2_32.dir/src/dllmain.c.obj 
2024-05-13T09:02:49.7646065Z C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\HostX64\x64\cl.exe  /nologo -DDBG=1 -DDLL_EXPORT_VERSION=0x600 -DLE -DMINGW_HAS_SECURE_API=1 -DUSE_COMPILER_EXCEPTIONS -DWINVER=0x502 -D_AMD64_ -D_CRT_NON_CONFORMING_SWPRINTFS -D_DLL -D_M_AMD64 -D_NEW_DELETE_OPERATORS_ -D_SEH_ENABLE_TRACE -D_SETUPAPI_VER=0x502 -D_WIN32_IE=0x600 -D_WIN32_WINDOWS=0x502 -D_WIN32_WINNT=0x502 -D_WIN64 -D__REACTOS__ -D__RELFILE__="&__FILE__[__FILE__[0] == '.' ? 6 : 25]" -D__RUNTIME_CHECKS__ -D__STDC__=1 -D__USE_CRTIMP -D__x86_64 -D__x86_64__ -ID:\a\reactos\reactos\build\dll\win32\ws2_32 -ID:\a\reactos\reactos\src\dll\win32\ws2_32 -ID:\a\reactos\reactos\src\sdk\include -ID:\a\reactos\reactos\src\sdk\include\psdk -ID:\a\reactos\reactos\src\sdk\include\dxsdk -ID:\a\reactos\reactos\build\sdk\include -ID:\a\reactos\reactos\build\sdk\include\psdk -ID:\a\reactos\reactos\build\sdk\include\dxsdk -ID:\a\reactos\reactos\build\sdk\include\ddk -ID:\a\reactos\reactos\build\sdk\include\reactos -ID:\a\reactos\reactos\build\sdk\include\reactos\mc -ID:\a\reactos\reactos\src\sdk\include\crt -ID:\a\reactos\reactos\src\sdk\include\ddk -ID:\a\reactos\reactos\src\sdk\include\ndk -ID:\a\reactos\reactos\src\sdk\include\reactos -ID:\a\reactos\reactos\src\sdk\include\reactos\libs -ID:\a\reactos\reactos\src\dll\win32\ws2_32\inc -ID:\a\reactos\reactos\src\sdk\include\reactos\winsock -ID:\a\reactos\reactos\src\sdk\lib\pseh\include /Ob0 /Od /GF /Gy /X /Zl /GS- /FS /Zc:threadSafeInit- /d2FH4- /W3 /wd4244 /wd4290 /wd4800 /wd4200 /wd4214 /wd4018 /we4013 /we4020 /we4022 /we4028 /we4047 /we4090 /we4098 /we4113 /we4129 /we4133 /we4163 /we4229 /we4311 /we4312 /we4313 /we4477 /we4603 /we4700 /we4715 /we4716 /we4101 /we4189 /w14115 /Zi /RTC1 /YuD:/a/reactos/reactos/build/dll/win32/ws2_32/CMakeFiles/ws2_32.dir/cmake_pch.h /FpD:/a/reactos/reactos/build/dll/win32/ws2_32/CMakeFiles/ws2_32.dir/./cmake_pch.c.pch /FID:/a/reactos/reactos/build/dll/win32/ws2_32/CMakeFiles/ws2_32.dir/cmake_pch.h /showIncludes /Fodll\win32\ws2_32\CMakeFiles\ws2_32.dir\src\dllmain.c.obj /Fddll\win32\ws2_32\CMakeFiles\ws2_32.dir\ /FS -c D:\a\reactos\reactos\src\dll\win32\ws2_32\src\dllmain.c
2024-05-13T09:02:49.7663767Z D:\a\reactos\reactos\src\dll\win32\ws2_32\src\dllmain.c(128): fatal error C1051: program database file, 'D:\a\reactos\reactos\build\dll\win32\ws2_32\CMakeFiles\ws2_32.dir\vc140.pdb', has an obsolete format, delete it and recompile
2024-05-13T09:02:49.7667459Z [8440/13469] Building C object dll\win32\wshirda\CMakeFiles\wshirda.dir\wshirda.c.obj
2024-05-13T09:02:49.7670134Z [8441/13469] Building C object dll\win32\wshom.ocx\CMakeFiles\wshom.dir\guid.c.obj
2024-05-13T09:02:49.7675509Z FAILED: dll/win32/wshom.ocx/CMakeFiles/wshom.dir/guid.c.obj 
2024-05-13T09:02:49.7924268Z C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\HostX64\x64\cl.exe  /nologo -DDBG=1 -DDLL_EXPORT_VERSION=0x600 -DMINGW_HAS_SECURE_API=1 -DUSE_COMPILER_EXCEPTIONS -DWINVER=0x502 -D_AMD64_ -D_CRT_NON_CONFORMING_SWPRINTFS -D_DLL -D_M_AMD64 -D_NEW_DELETE_OPERATORS_ -D_SEH_ENABLE_TRACE -D_SETUPAPI_VER=0x502 -D_WIN32_IE=0x600 -D_WIN32_WINDOWS=0x502 -D_WIN32_WINNT=0x600 -D_WIN64 -D__REACTOS__ -D__RELFILE__="&__FILE__[__FILE__[0] == '.' ? 6 : 25]" -D__RUNTIME_CHECKS__ -D__STDC__=1 -D__USE_CRTIMP -D__WINESRC__ -D__x86_64 -D__x86_64__ -ID:\a\reactos\reactos\build\dll\win32\wshom.ocx -ID:\a\reactos\reactos\src\dll\win32\wshom.ocx -ID:\a\reactos\reactos\src\sdk\include -ID:\a\reactos\reactos\src\sdk\include\psdk -ID:\a\reactos\reactos\src\sdk\include\dxsdk -ID:\a\reactos\reactos\build\sdk\include -ID:\a\reactos\reactos\build\sdk\include\psdk -ID:\a\reactos\reactos\build\sdk\include\dxsdk -ID:\a\reactos\reactos\build\sdk\include\ddk -ID:\a\reactos\reactos\build\sdk\include\reactos -ID:\a\reactos\reactos\build\sdk\include\reactos\mc -ID:\a\reactos\reactos\src\sdk\include\crt -ID:\a\reactos\reactos\src\sdk\include\ddk -ID:\a\reactos\reactos\src\sdk\include\ndk -ID:\a\reactos\reactos\src\sdk\include\reactos -ID:\a\reactos\reactos\src\sdk\include\reactos\libs -ID:\a\reactos\reactos\src\sdk\include\reactos\wine /Ob0 /Od /GF /Gy /X /Zl /GS- /FS /Zc:threadSafeInit- /d2FH4- /W3 /wd4244 /wd4290 /wd4800 /wd4200 /wd4214 /wd4018 /we4013 /we4020 /we4022 /we4028 /we4047 /we4090 /we4098 /we4113 /we4129 /we4133 /we4163 /we4229 /we4311 /we4312 /we4313 /we4477 /we4603 /we4700 /we4715 /we4716 /we4101 /we4189 /w14115 /Zi /RTC1 /showIncludes /Fodll\win32\wshom.ocx\CMakeFiles\wshom.dir\guid.c.obj /Fddll\win32\wshom.ocx\CMakeFiles\wshom.dir\ /FS -c D:\a\reactos\reactos\src\dll\win32\wshom.ocx\guid.c
2024-05-13T09:02:49.8066711Z D:\a\reactos\reactos\src\dll\win32\wshom.ocx\guid.c(16): fatal error C1090: PDB API call failed, error code '23': (0x000006BA)
2024-05-13T09:02:49.8085836Z [8442/13469] Building RC object dll\win32\wshom.ocx\CMakeFiles\wshom.dir\wshom.rc.res
2024-05-13T09:02:49.8091187Z [8443/13469] Building C object dll\win32\wshtcpip\CMakeFiles\wshtcpip.dir\iflist.c.obj
2024-05-13T09:02:49.8138032Z FAILED: dll/win32/wshtcpip/CMakeFiles/wshtcpip.dir/iflist.c.obj 
2024-05-13T09:02:49.8191749Z C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\HostX64\x64\cl.exe  /nologo -DDBG=1 -DDLL_EXPORT_VERSION=0x600 -DMINGW_HAS_SECURE_API=1 -DUNICODE -DUSE_COMPILER_EXCEPTIONS -DWINVER=0x502 -D_AMD64_ -D_CRT_NON_CONFORMING_SWPRINTFS -D_DLL -D_M_AMD64 -D_NEW_DELETE_OPERATORS_ -D_SEH_ENABLE_TRACE -D_SETUPAPI_VER=0x502 -D_UNICODE -D_WIN32_IE=0x600 -D_WIN32_WINDOWS=0x502 -D_WIN32_WINNT=0x502 -D_WIN64 -D__REACTOS__ -D__RELFILE__="&__FILE__[__FILE__[0] == '.' ? 6 : 25]" -D__RUNTIME_CHECKS__ -D__STDC__=1 -D__USE_CRTIMP -D__x86_64 -D__x86_64__ -ID:\a\reactos\reactos\build\dll\win32\wshtcpip -ID:\a\reactos\reactos\src\dll\win32\wshtcpip -ID:\a\reactos\reactos\src\sdk\include -ID:\a\reactos\reactos\src\sdk\include\psdk -ID:\a\reactos\reactos\src\sdk\include\dxsdk -ID:\a\reactos\reactos\build\sdk\include -ID:\a\reactos\reactos\build\sdk\include\psdk -ID:\a\reactos\reactos\build\sdk\include\dxsdk -ID:\a\reactos\reactos\build\sdk\include\ddk -ID:\a\reactos\reactos\build\sdk\include\reactos -ID:\a\reactos\reactos\build\sdk\include\reactos\mc -ID:\a\reactos\reactos\src\sdk\include\crt -ID:\a\reactos\reactos\src\sdk\include\ddk -ID:\a\reactos\reactos\src\sdk\include\ndk -ID:\a\reactos\reactos\src\sdk\include\reactos -ID:\a\reactos\reactos\src\sdk\include\reactos\libs -ID:\a\reactos\reactos\src\sdk\lib\tdilib /Ob0 /Od /GF /Gy /X /Zl /GS- /FS /Zc:threadSafeInit- /d2FH4- /W3 /wd4244 /wd4290 /wd4800 /wd4200 /wd4214 /wd4018 /we4013 /we4020 /we4022 /we4028 /we4047 /we4090 /we4098 /we4113 /we4129 /we4133 /we4163 /we4229 /we4311 /we4312 /we4313 /we4477 /we4603 /we4700 /we4715 /we4716 /we4101 /we4189 /w14115 /Zi /RTC1 /showIncludes /Fodll\win32\wshtcpip\CMakeFiles\wshtcpip.dir\iflist.c.obj /Fddll\win32\wshtcpip\CMakeFiles\wshtcpip.dir\ /FS -c D:\a\reactos\reactos\src\dll\win32\wshtcpip\iflist.c
2024-05-13T09:02:49.8206145Z D:\a\reactos\reactos\src\dll\win32\wshtcpip\iflist.c(442): fatal error C1090: PDB API call failed, error code '23': (0x000006BA)
2024-05-13T09:02:49.8217312Z [8444/13469] Building C object dll\win32\wshom.ocx\CMakeFiles\wshom.dir\shell.c.obj
2024-05-13T09:02:49.8265160Z FAILED: dll/win32/wshom.ocx/CMakeFiles/wshom.dir/shell.c.obj 
2024-05-13T09:02:49.8393710Z C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\HostX64\x64\cl.exe  /nologo -DDBG=1 -DDLL_EXPORT_VERSION=0x600 -DMINGW_HAS_SECURE_API=1 -DUSE_COMPILER_EXCEPTIONS -DWINVER=0x502 -D_AMD64_ -D_CRT_NON_CONFORMING_SWPRINTFS -D_DLL -D_M_AMD64 -D_NEW_DELETE_OPERATORS_ -D_SEH_ENABLE_TRACE -D_SETUPAPI_VER=0x502 -D_WIN32_IE=0x600 -D_WIN32_WINDOWS=0x502 -D_WIN32_WINNT=0x600 -D_WIN64 -D__REACTOS__ -D__RELFILE__="&__FILE__[__FILE__[0] == '.' ? 6 : 25]" -D__RUNTIME_CHECKS__ -D__STDC__=1 -D__USE_CRTIMP -D__WINESRC__ -D__x86_64 -D__x86_64__ -ID:\a\reactos\reactos\build\dll\win32\wshom.ocx -ID:\a\reactos\reactos\src\dll\win32\wshom.ocx -ID:\a\reactos\reactos\src\sdk\include -ID:\a\reactos\reactos\src\sdk\include\psdk -ID:\a\reactos\reactos\src\sdk\include\dxsdk -ID:\a\reactos\reactos\build\sdk\include -ID:\a\reactos\reactos\build\sdk\include\psdk -ID:\a\reactos\reactos\build\sdk\include\dxsdk -ID:\a\reactos\reactos\build\sdk\include\ddk -ID:\a\reactos\reactos\build\sdk\include\reactos -ID:\a\reactos\reactos\build\sdk\include\reactos\mc -ID:\a\reactos\reactos\src\sdk\include\crt -ID:\a\reactos\reactos\src\sdk\include\ddk -ID:\a\reactos\reactos\src\sdk\include\ndk -ID:\a\reactos\reactos\src\sdk\include\reactos -ID:\a\reactos\reactos\src\sdk\include\reactos\libs -ID:\a\reactos\reactos\src\sdk\include\reactos\wine /Ob0 /Od /GF /Gy /X /Zl /GS- /FS /Zc:threadSafeInit- /d2FH4- /W3 /wd4244 /wd4290 /wd4800 /wd4200 /wd4214 /wd4018 /we4013 /we4020 /we4022 /we4028 /we4047 /we4090 /we4098 /we4113 /we4129 /we4133 /we4163 /we4229 /we4311 /we4312 /we4313 /we4477 /we4603 /we4700 /we4715 /we4716 /we4101 /we4189 /w14115 /Zi /RTC1 /YuD:/a/reactos/reactos/build/dll/win32/wshom.ocx/CMakeFiles/wshom.dir/cmake_pch.h /FpD:/a/reactos/reactos/build/dll/win32/wshom.ocx/CMakeFiles/wshom.dir/./cmake_pch.c.pch /FID:/a/reactos/reactos/build/dll/win32/wshom.ocx/CMakeFiles/wshom.dir/cmake_pch.h /showIncludes /Fodll\win32\wshom.ocx\CMakeFiles\wshom.dir\shell.c.obj /Fddll\win32\wshom.ocx\CMakeFiles\wshom.dir\ /FS -c D:\a\reactos\reactos\src\dll\win32\wshom.ocx\shell.c
2024-05-13T09:02:49.8410688Z D:\a\reactos\reactos\src\dll\win32\wshom.ocx\shell.c(1833): fatal error C1090: PDB API call failed, error code '23': (0x00000006)
```

Try again...

**Review by JoachimHenze — APPROVED** [review]
We are almost there: latest gcc x86 dbgNT502
image

I am not sure why it does still complain about those 2 since they are now both mentioned in the GCC baseaddress.cmake.
And in releases/0.4.7 to releases/0.4.14 that works and is enough, and leaves us with zero warnings in this context.

In the old branches I had a final empty-line in the file.... Not sure whether that helps to cut off one more... ???
image
That's a pic from releases/0.4.11 for example, see the blinking cursor...


But I am now giving my approval already, because the status after your PR is now already definitely better than the status of master head.


For MSVC dbg, the current state warns about a few more modules still, but I don't remember what the status quo on master head is currently with that, because I cannot build with that myself anymore since the VS2010-dropping:
image

**Pull request merged**